packages/openapi3/src/openapi.ts

```diff
@@ -797,11 +797,16 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
     if (type === "model" || type === "array") {
       if (nonNullOptions.length === 1) {
         // Get the schema for the model type
-        const schema: any = getSchemaForType(nonNullOptions[0]);
-        if (nullable) {
-          schema["nullable"] = true;
+        let schema: any = getSchemaOrRef(nonNullOptions[0]);
+        if (nullable && schema?.$ref) {
```

Comment:
```suggestion
        if (nullable && schema.$ref) {
```
is the `?` needed? feel like it will crash in the `else` statement if it actually is `undefined` sometimes